components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2162,6 +2181,60 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
         return appInfo.toArray(new ApplicationBasicInfo[0]);
     }
 
+    /**
+     * Get query to exclude system portals if excludeSystemPortals is true.
+     *
+     * @param excludeSystemPortals Exclude system portals.
+     * @return
+     */
+    private String populateSystemPortalsExcludeQuery(Boolean excludeSystemPortals) {
+
+        return excludeSystemPortals ?
+                String.format(ApplicationMgtDBQueries.EXCLUDE_SYSTEM_PORTALS_BY_NAME,
+                        getApplications(SYSTEM_PORTALS).stream()
+                                .map(s -> "'" + s + "'")
+                                .collect(Collectors.joining(", ")))
+                : "";
+    }
+
+    /**
+     * Get system portals from configuration.
+     *
+     * @param parentElement
+     * @return
+     */
+    private static Set<String> getApplications(String parentElement) {
+
+        IdentityConfigParser configParser = IdentityConfigParser.getInstance();
+        OMElement systemApplicationsConfig = configParser.getConfigElement(parentElement);
+        if (systemApplicationsConfig == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("'" + parentElement + "' config not found.");
+            }
+            return Collections.emptySet();
+        }
+
+        Iterator applicationIdentifierIterator = systemApplicationsConfig
+                .getChildrenWithLocalName(APPLICATION_NAME_CONFIG_ELEMENT);
+        if (applicationIdentifierIterator == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("'" + APPLICATION_NAME_CONFIG_ELEMENT + "' config not found.");
+            }
+            return Collections.emptySet();
+        }
+
+        Set<String> applications = new HashSet<>();
+        while (applicationIdentifierIterator.hasNext()) {
+            OMElement applicationIdentifierConfig = (OMElement) applicationIdentifierIterator.next();
+            String applicationName = applicationIdentifierConfig.getText();
+            if (StringUtils.isNotBlank(applicationName)) {
+                applications.add(applicationName.trim());
+            }
+        }
+
```

Comment:
Remove this line.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/964c6d3814dba4a93591240fb7b95b4f3b9f77c3